protocol/messenger.go

```diff
@@ -2047,7 +2022,23 @@ func (m *Messenger) MarkMessagesSeen(chatID string, ids []string) error {
 	return nil
 }
 
-// DEPRECATED: required by status-react.
+func (m *Messenger) MarkAllRead(chatID string) error {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	chat, ok := m.allChats[chatID]
+	if !ok {
+		return errors.New("chat not found")
+	}
+
+	err := m.persistence.MarkAllRead(chatID)
```

Comment:
Marking all as read and saving the chat is not an atomic operation. Is this ok?

Reply:
Worth making it atomic I guess, will change that